vyper/builtin_functions/functions.py

```diff
@@ -290,6 +290,32 @@ def build_LLL(self, expr, args, kwargs, context):
             ]
             return LLLnode.from_list(node, typ=ByteArrayType(length.value), location="memory")
 
+        # special handling for slice(<address>.code, start, length)
+        if sub.value == "~extcode":
+            # `length` is constant (validated by `validate_address_code_attribute`)
+            assert isinstance(length.value, int)
+            address = sub.args[0]
+            if address.value == "address":  # self.code
+                node = [
+                    "seq",
+                    ["mstore", np, length],
+                    ["codecopy", np + 32, start, length],
+                    np,
+                ]
+            else:
+                node = [
+                    "with",
```

Comment:
I don't think that this needs to be cached into the `_extcode_address` variable, since it is only referenced once in the enclosed block

Reply:
Thanks for spotting it.
Initially, I was using this in two places, one for bound check with `extcodesize` and another for `extcodecopy`.
Then, I removed the bound check.

As per your comment https://github.com/vyperlang/vyper/pull/2583#issuecomment-1004202154, I bought back the bound check, thus I think it's good to have `_extcode_address` variable as well https://github.com/vyperlang/vyper/pull/2583/commits/76fce00ad88b5d7919dec4c4f3e90bd6d194cd43.